docs/internals/variable_cleanup.rst

```diff
@@ -1,29 +1,31 @@
 .. index: variable cleanup
 
 *********************
-Cleaning Up Variables
+Zeroing of Padding
 *********************
 
-When a value is shorter than 256 bit, in some cases the remaining bits
-must be cleaned.
-The Solidity compiler is designed to clean such remaining bits before any operations
-that might be adversely affected by the potential garbage in the remaining bits.
-For example, before writing a value to  memory, the remaining bits need
-to be cleared because the memory contents can be used for computing
-hashes or sent as the data of a message call.  Similarly, before
-storing a value in the storage, the remaining bits need to be cleaned
-because otherwise the garbled value can be observed.
+When a value is shorter than 256 bit, in some cases the remaining (padding) bits
+must be zeroed. The Solidity compiler is designed to zero any padding bits before operations
```

Comment:
Well, the word "cleaned" was ambiguous to me. I was thinking, did it mean "zero" or did it mean sign extend?

The reason I went on this excursion is that the original text contains this statement: "the remaining bits need to be cleared because the memory contents can be used for computing hashes"

Note the word "cleared", which maybe it was meant to read "cleaned".

I can edit this documentation to clarify what is meant by "cleaned" if it means sign-extend even in the context of computing hashes. Please advise.

Reply:
It should indeed say ``cleaned`` instead of ``cleared``.
The table in the end of this section should explain, what "cleaning" means, though  (not that things cannot be improved to make this clearer).